### ISSUE TITLE
train on object365

### ISSUE THREAD

**Issue by lleye** [ultralytics/yolov5#3392]
i am trying to train on object365 data from scratch. should i use the [hyp.finetune_objects365.yaml](https://github.com/ultralytics/yolov5/blob/develop/data/hyp.finetune_objects365.yaml) file instead of the [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/develop/data/hyp.scratch.yaml)?

i am starting from scratch. also did you do evolution to get these hyperparameter settings?

**Comment by glenn-jocher** [comment]
@lleye you'd probably want to ask the original author of those two files. See PR https://github.com/ultralytics/yolov5/pull/2932. 

Also note objects365 will autodownload and convert to YOLOv5 format, but we still need to establish an 'official' train/val split using out autosplit function to create two txt file lists of images:
https://github.com/ultralytics/yolov5/blob/fdbe527dc02f6f8891a1fd0baa3c5638ed5f53a4/utils/datasets.py#L1051-L1058

**Comment by lleye** [comment]
@ferdinandl007 any thoughts on above question

**Comment by ferdinandl007** [comment]
@lleye use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine tuning the model for the fastest conversions.

The hyper parameters were found by evolution about 350 Evolutions.

**Comment by lleye** [comment]
thanks @ferdinandl007 I tried (50 epochs on hyp.scratch then 50 on hyp. finetune_objects365) and got a small improvement ~1%. do you have any other suggestions on improving mAP?

**Comment by ferdinandl007** [comment]
@lleye, what's your current APm0.5, on your evaluation set?
What resolution are you training on?
And what version of YOLOv5?
Objects 360 has a lot of very small objects so larger resolutions can greatly increase your APm score!
Additionally I noticed that increasing the  Batch size to max out your GPU’s memory, helps the model to generalise, for fine-tuning I got pretty good results with a Batch size of 180 and above
Another technique you can try is identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!
Also by default object 365 has crowd boxes, which can be good if you want to detect such things, but can greatly reduce your APm score. So you might want to ignore images with crowd boxes or block them out somehow by filling them in with a solid colour or something like that!

**Comment by lleye** [comment]
@ferdinandl007 sorry for late reply, mAP 0.5 is ~.18 using your approach though i use other training/val data too. Not sure how to mask out the crowd annotations, is there a iscrowd flag to indicate that?

Do you also have any other suggestions? i saw on the other page you have like .45 though you remove the crowd annotations

**Comment by glenn-jocher** [comment]
@lleye the default Objects365.yaml autodownloader already removes the crowd annotations:
https://github.com/ultralytics/yolov5/blob/9d86b54eb37ea9d2b4ae3cda0dfe8ab7aa16f2c6/data/Objects365.yaml#L95-L96

**Comment by Auth0rM0rgan** [comment]
Hey @glenn-jocher, 

I have a question about the object365 dataset. I used the following command `python train.py --data Objects365.yaml` to download the dataset and train yolov5 with it. But after downloading and everything, the dataset folder only contains a JSON file and empty folders for image and label. There are not any images and only train labels exist in the dataset folder. Should I download the images by myself? can you help me to understand how to train yolov5 on object365?

Thanks!

**Comment by glenn-jocher** [comment]
@Auth0rM0rgan Objects365 is a massive dataset with several hundred GB of multiple zip files which may take a day or more to completely download even if you have a superfast internet connection. If your connection to the servers in China is poor you are likely not receiving the data correctly.

**Comment by tucker666** [comment]
> @lleye, what's your current APm0.5, on your evaluation set?
> What resolution are you training on?
> And what version of YOLOv5?
> Objects 360 has a lot of very small objects so larger resolutions can greatly increase your APm score!
> Additionally I noticed that increasing the Batch size to max out your GPU’s memory, helps the model to generalise, for fine-tuning I got pretty good results with a Batch size of 180 and above
> Another technique you can try is identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!
> Also by default object 365 has crowd boxes, which can be good if you want to detect such things, but can greatly reduce your APm score. So you might want to ignore images with crowd boxes or block them out somehow by filling them in with a solid colour or something like that!

hello, thanks for your PR, and i am training yolov5s on objects365 at 640 resolution, now objects365 has its official val dataset on website (wget "https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365%E6%95%B0%E6%8D%AE%E9%9B%86/val/zhiyuan_objv2_val.json") and i got a mAP50 about ~0.19, 
I wonder which model  and what image resolution did you use, and how much was the mAP50 ? thank you very much

**Comment by glenn-jocher** [comment]
@tucker666 thanks for the info! Can you submit a PR with an update to Objects365.yaml with the new val split? The file you'd want to modify is here, in the `download:` field:

https://github.com/ultralytics/yolov5/blob/951922c735c1e98b596fd9845de25a62fcdc7c73/data/Objects365.yaml#L60-L103

Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.

**Comment by tycallen** [comment]
> @lleye use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine tuning the model for the fastest conversions.
> 
> The hyper parameters were found by evolution about 350 Evolutions.

how many epoch should be in the finetune? 300 is a little too much for me

**Comment by lleye** [comment]
@tycallen the longer the epochs the better but significant diminishing returns after 150

**Comment by wangsun1996** [comment]
> > @lleye, what's your current APm0.5, on your evaluation set?
> > What resolution are you training on?
> > And what version of YOLOv5?
> > Objects 360 has a lot of very small objects so larger resolutions can greatly increase your APm score!
> > Additionally I noticed that increasing the Batch size to max out your GPU’s memory, helps the model to generalise, for fine-tuning I got pretty good results with a Batch size of 180 and above
> > Another technique you can try is identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!
> > Also by default object 365 has crowd boxes, which can be good if you want to detect such things, but can greatly reduce your APm score. So you might want to ignore images with crowd boxes or block them out somehow by filling them in with a solid colour or something like that!
> 
> hello, thanks for your PR, and i am training yolov5s on objects365 at 640 resolution, now objects365 has its official val dataset on website (wget "https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365%E6%95%B0%E6%8D%AE%E9%9B%86/val/zhiyuan_objv2_val.json") and i got a mAP50 about ~0.19, I wonder which model and what image resolution did you use, and how much was the mAP50 ? thank you very much

could you provide a  yolov5s.pt on obj365?thank you very much!

**Comment by wangsun1996** [comment]
> @lleye use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine tuning the model for the fastest conversions.
> 
> The hyper parameters were found by evolution about 350 Evolutions.
could you provide any yolov5.pt on obj365(in addition to yolov5m.pt on obj365)?thank you very much!

**Comment by wangsun1996** [comment]
> > @lleye, what's your current APm0.5, on your evaluation set?
> > What resolution are you training on?
> > And what version of YOLOv5?
> > Objects 360 has a lot of very small objects so larger resolutions can greatly increase your APm score!
> > Additionally I noticed that increasing the Batch size to max out your GPU’s memory, helps the model to generalise, for fine-tuning I got pretty good results with a Batch size of 180 and above
> > Another technique you can try is identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!
> > Also by default object 365 has crowd boxes, which can be good if you want to detect such things, but can greatly reduce your APm score. So you might want to ignore images with crowd boxes or block them out somehow by filling them in with a solid colour or something like that!
> 
> hello, thanks for your PR, and i am training yolov5s on objects365 at 640 resolution, now objects365 has its official val dataset on website (wget "https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365%E6%95%B0%E6%8D%AE%E9%9B%86/val/zhiyuan_objv2_val.json") and i got a mAP50 about ~0.19, I wonder which model and what image resolution did you use, and how much was the mAP50 ? thank you very much



> @lleye use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine tuning the model for the fastest conversions.
> 
> The hyper parameters were found by evolution about 350 Evolutions.

Could you provide any yolov5.pt on obj365(in addition to yolov5m.pt on obj365)? Thank you very much!

**Comment by wangsun1996** [comment]
> @lleye, what's your current APm0.5, on your evaluation set? What resolution are you training on? And what version of YOLOv5? Objects 360 has a lot of very small objects so larger resolutions can greatly increase your APm score! Additionally I noticed that increasing the Batch size to max out your GPU’s memory, helps the model to generalise, for fine-tuning I got pretty good results with a Batch size of 180 and above Another technique you can try is identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so! Also by default object 365 has crowd boxes, which can be good if you want to detect such things, but can greatly reduce your APm score. So you might want to ignore images with crowd boxes or block them out somehow by filling them in with a solid colour or something like that!

Further, when you train yolov5 on obj365,  did you use pretrained model(such as yolov5s.pt on COCO dataset? )?

**Comment by wangsun1996** [comment]
> > > @lleye, what's your current APm0.5, on your evaluation set?
> > > What resolution are you training on?
> > > And what version of YOLOv5?
> > > Objects 360 has a lot of very small objects so larger resolutions can greatly increase your APm score!
> > > Additionally I noticed that increasing the Batch size to max out your GPU’s memory, helps the model to generalise, for fine-tuning I got pretty good results with a Batch size of 180 and above
> > > Another technique you can try is identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!
> > > Also by default object 365 has crowd boxes, which can be good if you want to detect such things, but can greatly reduce your APm score. So you might want to ignore images with crowd boxes or block them out somehow by filling them in with a solid colour or something like that!
> > 
> > 
> > hello, thanks for your PR, and i am training yolov5s on objects365 at 640 resolution, now objects365 has its official val dataset on website (wget "https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365%E6%95%B0%E6%8D%AE%E9%9B%86/val/zhiyuan_objv2_val.json") and i got a mAP50 about ~0.19, I wonder which model and what image resolution did you use, and how much was the mAP50 ? thank you very much
> 
> > @lleye use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine tuning the model for the fastest conversions.
> > The hyper parameters were found by evolution about 350 Evolutions.
> 
> Could you provide any yolov5.pt on obj365(in addition to yolov5m.pt on obj365)? Thank you very much!

such as yolov5s.pt or yolov5s6.pt on obj365

**Comment by glenn-jocher** [comment]
https://github.com/ultralytics/yolov5/releases/download/v6.0/yolov5m_Objects365.pt

**Comment by zhangby2085** [comment]
> https://github.com/ultralytics/yolov5/releases/download/v6.0/yolov5m_Objects365.pt

Thank you for above weight file, is it better compare to the coco trained yolov5m.pt? the transfer learning perspective, and do you have yolov5x_Object365.pt, it gonna takes years for me

**Comment by tino926** [comment]
Hi, @ferdinandl007 

can you share how you achieve theses?
1. "use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine tuning the model"  
2. "identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!"  

For 1., did you train with hyp.scratch.yaml and set epochs to 20, then use the weights as initial weights and start another training with hyp.finetune_objects365.yaml?

For 2., does the "10 iterations" means: after a full training, start another training with only 10 epochs? Another question is, how to let yolov5 train on part of classes? did you modify the dataloader?

**Comment by glenn-jocher** [comment]
Hi @tino926,

To achieve the process described:

1. "use hyp.scratch.yaml for the first 20-50 Epoque and then use hyp.finetune_objects365.yaml for fine-tuning the model": 
You can start by training the model with hyp.scratch.yaml for the specified number of epochs (20-50). Once those initial epochs are completed, you can use the weights obtained from this training as the initial weights for another training session using hyp.finetune_objects365.yaml. This process allows for fine-tuning the model on the specific dataset.

2. "identifying the classes you have the greatest loss on, and train on the classes exclusively for 10 iterations or so!":
To identify the classes with the greatest loss, you can analyze the training loss values for each class during training. Once you have identified the classes, you can train the model exclusively on those selected classes for a shorter duration, such as 10 iterations or epochs. This focused training on specific classes can help improve the model's performance on those particular classes. You may need to modify the dataloader to extract and load only the data related to the selected classes during this specific training phase.

Hope this helps! Let me know if you have any further questions.

Glenn
